### PR TITLE
[#1495] Hopefully tests are more stable now...

### PR DESCRIPTION
**Pull request by jung-kim**
- fix click tests!! hooray!!!
- Using `./source/utils/logger.js`, this slimmer and less dependent logger now helps us see browser logs at click tests.
- removed many of the blind wait timeouts and replaced them with `waitForNetworkIdle()` (this helped out so much and fixed issues where an element is replaced in between dom detection and dom click.
- added `triggerProgramEvents()` to manually trigger program events after certain operations.  (this is because certain operations have race conditions where UI fails detects changes because events can trigger before server side can detect it.  In real life situations, this is never a issue because mouse movement triggers these events and corrects itself if events are missed)
- added `portfinder` dependency rather than inhouse port finding logics.

---

This was so annoying because it passed on my local comp and had to push every single changes to check if passes on github actions... 

Also, I'm sure there are few more twicks needed as we go...

### CODE REVIEW

File: clicktests/spec.discard.js [hunk 1]
```diff
@@ -70,10 +82,16 @@ describe('[DISCARD - withWarn]', () => {
     return createAndDiscard(environment, testRepoPaths[0], 'yes');
   });
 
-  it('Should be possible to discard a created file and disable warn for awhile', async () => {
+  it('Should be possible to discard a created file and disable warn for awhile', async function () {
```
Comment (campersau) [inline clicktests/spec.discard.js:85]: Why `function`?

```suggestion
  it('Should be possible to discard a created file and disable warn for awhile', async () {
```

**Review by campersau — COMMENTED** [review]
Hm builds take around 3-5 minutes longer now. In some cases it is 2x [before](https://ci.appveyor.com/project/FredrikNoren/ungit/builds/41926979) [after](https://ci.appveyor.com/project/FredrikNoren/ungit/builds/41962046)

File: public/source/server.js [hunk 2]
```diff
@@ -178,6 +180,8 @@ Server.prototype.putPromise = function (url, arg) {
 };
 
 Server.prototype.unhandledRejection = function (err) {
+  const stacktrace = Error().stack;
```
[the remark this answers is not in the document]
Comment (jung-kim) [reply public/source/server.js:183]: I think so, `unhandledRejection` is bit tricky to track down what happened and it is nice to see the stacktraces up to this function call to see how it got here

**Comment by jung-kim** [comment]
Admittedly, tests are slower.  A lot of coordinations are added that slowed down.  But honestly, tests on the local comp is not too bad.

Hopefully, this PR is to bring stability to the wobbly tests.  And hopefully we can make it faster and more efficient.  Maybe with pausing the program events triggers thus preventing potential dom changes, which seems to be the main reason why tests are so unstable.

**Pull request merged**